storage/badger/config.go

```diff
@@ -107,6 +106,8 @@ func WithTruncate(trunc bool) Opt {
 
 func WithPersist(persist bool) Opt {
 	return func(c *Config) {
-		c.InMemory = c.InMemory && !persist
+		if persist {
+			c.InMemory = false
+		}
```

Comment:
Could this be `c.InMemory = !persist` ?

Reply:
@DylanTinianov that was the bug actually, I thought it could be. 

But turns out it should have to be only in one direction ( because it conflicts with snapshot )

Reply:
Ahh I see